packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -133,9 +139,9 @@ export const useTriggerElement = (state: MenuTriggerState): MenuTriggerState =>
         }),
   };
 
-  state.children = React.cloneElement(child, {
+  state.children = applyTriggerPropsToChildren<MenuTriggerChildProps & { ref?: React.Ref<unknown> }>(state.children, {
```

Comment:
`MenuTriggerChildProps` should include `ref` in the type, so it's in the type passed to the render function.

Also, I think TypeScript should automatically infer the type for `applyTriggerPropsToChildren` here, so you should be able to leave it out:
```suggestion
  state.children = applyTriggerPropsToChildren(state.children, {
```